Creator: object deselect is explicit, prevent accidental selection switch

dmitry-yudakov opened the pull request:
The purpose is to avoid accidental selection of different objects while in creator mode - so if you want to change function of another object, you need first to deselect with X then select again with click

It's temp disabled for gizmo because it seems to have some internal logic making it switch the object on click.

dmitry-yudakov committed:
feat(creator): object deselect is explicit, don't change selection on click if already selected except gizmo

The purpose is to avoid accidental selection of different objects while in creator mode - so if you want to change function of another object, you need first to deselect with X then select again with click

It's temp disabled for gizmo because it seems to have some internal logic making it switch the object on click.

PAKovalev commented on packages/app/src/stores/UniverseStore/models/World3dStore/World3dStore.ts:
```diff
@@ -57,6 +57,26 @@ const World3dStore = types
+    closeAndResetObjectMenu() {
+      console.log('closeAndResetObjectMenu', self.selectedObjectId);
+
+      const {creatorStore} = getRootStore(self);
```
const {creatorStore, widgetManagerStore} = getRootStore(self);
I don't like mixing stores. I may be under component to have proper tests for stores in the future )

dmitry-yudakov replied:
I agree. I'm thinking about moving this stuff into creatorStore and not have to pass it through world3dStore at all. Not sure about widgetManager though.

dmitry-yudakov committed:
feat(gizmo): prevent automatic object change on click

We want the gizmo behaviour to be predictable and controlled from main logic and not reattach automatically.

dmitry-yudakov committed:
Merge remote-tracking branch 'origin/develop' into feat/creator/prevent-deselection

dmitry-yudakov committed:
Merge remote-tracking branch 'origin/develop' into feat/creator/prevent-deselection
